src/selectors/entities/channel_categories.ts

```diff
@@ -189,6 +189,98 @@ export function makeFilterAutoclosedDMs(getAutocloseCutoff = getDefaultAutoclose
     );
 }
 
+export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        getCurrentChannelId,
+        (state: GlobalState) => state.entities.users.profiles,
+        getCurrentUserId,
+        getMyChannelMemberships,
+        (state: GlobalState) => getInt(state, Preferences.CATEGORY_SIDEBAR_SETTINGS, Preferences.LIMIT_VISIBLE_DMS_GMS),
+        (originalChannels, categoryType, currentChannelId, profiles, currentUserId, myMembers, limitPref) => {
+            if (categoryType !== CategoryTypes.DIRECT_MESSAGES) {
+                // Only autoclose DMs that haven't been assigned to a category
+                return originalChannels;
+            }
+
+            const channels = originalChannels.slice();
+
+            function sortByLastViewedAt(channelA: Channel, channelB: Channel) {
+                const channelALastViewedAt = myMembers[channelA.id]?.last_viewed_at || 0;
+                const channelBLastViewedAt = myMembers[channelB.id]?.last_viewed_at || 0;
+
+                if (channelBLastViewedAt === channelALastViewedAt) {
+                    return 0;
+                }
+
+                if (channelALastViewedAt === 0) {
```

Comment:
I don't think these cases are necessary since we'd want a channel that has never been viewed to be last. If it's something the user was just invited to, it should be unread, so that part of the sorting will handle it.

Reply:
If I remove this check it will sort the channels with last_viewed_at === 0 last and from my understanding we want to prioritise them.

Reply:
Is it possible to have a DM that has never been viewed open at all? Even one that was manually closed should have a `last_viewed_at` > 0.

Reply:
From what I understood brand new DM's will have a last_viewed_at === 0

Reply:
If another user creates it and sends a message, it will be 0 but it will be visible since it's unread. That's the only case I can think of where we'd have one like that naturally